frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -117,6 +188,15 @@ export function DynamicConfiguration () {
         }
     };
 
+    const addApplication = async (detail: any) => {
+        setSelectedApplicationOptions(detail.selectedOptions);
+        const newApps: Application[] = [];
```

Comment:
I would probably opt to update this name to `updatedSelectedApps` as "new" implies they are all new which isn't quite accurate here and you can use a [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) here too.

Seems like we do this kind of conversion frequently across the app (I think I remember doing this too). It would probably make sense to create a utility for this that we should be using across the application for `valueToOption` mapper and `optionToValue` mapper

Reply:
Nice idea, create a backlog task for it and we can tackle it across the app once we get app config out the door